### DISCUSSION TITLE
Compatibility: Duplicate Post plugins integration for Table Storage

### DISCUSSION THREAD

**Discussion by MrBin99** [pods-framework/pods#6597] [Ideas]
**Describe the bug**
I try to duplicate post from a custom post type created with Pods. But the problem is that no Pods fields are duplicated.
Wordpress default fields (such as title or featured image) and post metas related to the post are well duplicated but any field in the post form are filled (all empty or with the default value).
I use the plugin "Table Storage" for storing pods in different tables. When I duplicate a post I see the line created in the database but all columns are NULL.

I tried multiple plugin to duplicate posts, same result. Currently I use this one [https://fr.wordpress.org/plugins/post-duplicator/](https://fr.wordpress.org/plugins/post-duplicator/) 

**To Reproduce**
Go to a custom post type created via Pods and try to duplicate it.
It is well duplicated but with no fields filled with values of the other post.

**Expected behavior**
Duplicate fully the post with all custom fields from Pods.

**Screenshots**
![image](https://user-images.githubusercontent.com/9566430/62603201-b3875580-b8f5-11e9-81c7-ace27713fc5f.png)

![image](https://user-images.githubusercontent.com/9566430/62603260-d4e84180-b8f5-11e9-9a77-9302c08e39a2.png)

## Pods Version
2.7.14

## WordPress Environment
<details>
WordPress Version: 4.9.10

PHP Version: 7.0.27-0+deb9u1

MySQL Version: 5.5.5

Server Software: Apache/2.4.25 (Debian)

Your User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0

Currently Active Plugins:
    Classic Editor: 1.5
    Cookie Notice: 1.2.46
    DynamicConditions: 1.2.1
    Ele Custom Skin: 1.0.9
    Elementor: 2.5.16
    Elementor Pro: 2.5.8
    Enhanced Media Library: 2.7.2
    Gravity Forms: 2.4.6.2
    Hide Updates: 1.1.3
    Media File Renamer (Auto Rename): 4.5.4
    Ocean Elementor Widgets: 1.1.6
    Ocean Extra: 1.5.6
    Panda Pods Repeater Field: 1.4.2
    Plate: 3.0.2
    Pods - Custom Content Types and Fields: 2.7.14
    Post Duplicator: 2.20
    Smush: 3.2.0.1
    User Role Editor: 4.51
    WP Crontrol: 1.7.1
    WPS Hide Login: 1.5.2.1
    Yoast SEO: 9.7
</details>

**Possible Workaround**
In the method `save_pod_item` of the PodsAPI `$params->data` is an empty array so all the columns inserted in the database are NULL.

[a comment by an automated account is not shown]

**Comment by JoryHogeveen** [comment]
Hello @MrBin99,

I use this plugin: [Duplicate Post](https://nl.wordpress.org/plugins/duplicate-post/)
Never had any issues so this might be an issue related to the "Post Duplicator" plugin

**Comment by MrBin99** [comment]
Thanks @JoryHogeveen, now it works partially. Some fields are well duplicated but some are not.
In fact, only relationnal fields are copied but not plain fields (text, dates ...). In the database the rows are created but filled with NULL like before.

**Comment by JoryHogeveen** [comment]
Hello @MrBin99 
Sorry I cannot reproduce your issue. Just tested duplicating with a clean installation and all metadata is copied perfectly (attachment, date, wysiwyg and relationship).
I think this might be a setting issue or another plugin that is causing this issue.
Could you test if it works with only Pods and Duplicate Post enabled?

**Comment by MrBin99** [comment]
I've tested with a fresh install.

In fact, it doesn't work if you use the "Table Storage" plugin and if you set the pod to be stored in his own table.
When you duplicate, It creates the row in the database but no column are filled (all NULL).
If you set the pod to use post metas, to store fields, it works.

I think relationnal fields are stored in metas in both situation, this is why it works just for those fields. But plain fields are stored in the table newly created for the pod.

**Comment by JoryHogeveen** [comment]
Ah got it. Yes Table Storage won't be supported by any duplicate post plugin since this is a specific Pods feature. It's not stored as metadata so other plugins won't recognize it without custom integration.

**Comment by MrBin99** [comment]
Ok. But Pods functions for saving are called (I followed the trace). It's just that if it's a "Table Storage" pod, the plugin is not retreiving data from the custom table and do just the posts defaults and metas. This is why the empty row in the database, the save call is made but with no data :)

Do you plan to fix that ? 
If not do you know if there is maybe a Wordpress action or filter called before Pods save the row so I can use it to retreive the missing data and save the post ?

Thanks.

**Comment by JoryHogeveen** [comment]
Hi @MrBin99 

We might create some compatibility code in the future but since this is an edge case it won't be high priority, sorry. I'll keep this issue open for future reference.

You could always look for hooks in the duplicate posts plugin.

Cheers, Jory
